### ISSUE TITLE
Taiga Wallet API

### ISSUE THREAD

**Issue by bazzilic** [anoma/taiga#134]
Taiga API for wallets.

- [ ] `createPtx() : Ptx` Creates the initial partial transaction, spending user’s available assets and possibly creating some notes (intent notes or the desired assets directly). For every note created in the partial transaction, we need to include the nullifier key for the solvers to be able to spend them to balance the transaction
- [ ] `createEncData()` Actually because nullifier keys stored in the notes are short-term, we don’t need a separate call for that and can put it inside the createPtx() call. _Is this redundant then? @vveiln @XuyangSong_
- [ ] `retrieveVP()` Solvers need information about the VPs in order to be able to satisfy them efficiently. _There was some design question around that, was it resolved? @vveiln @XuyangSong_
- [ ] `createIntent()` Intent is a data structure that contains partial transactions, VPs and other data that might be required. Each intent contains at least one partial transaction.
- [ ] `finalizeTxn()` Trivial transactions (e.g., send tokens to someone) don’t need solving and can be finalized by the wallet itself thus saving solver fees.
- [ ] `ExecuteVP()` Solving/finalizing locally might require execution of VP. Does it need to be exposed or it will be concealed by `finalizeTxn()`? -- request for comments on this one @vveiln @XuyangSong @Acentelles

**Comment by XuyangSong** [comment]
> `createEncData()` Actually because nullifier keys stored in the notes are short-term, we don’t need a separate call for that and can put it inside the createPtx() call. Is this redundant then?

I agree. We might not need this API. We would use verifiable encryption in VP, and the encrypted data would be part of the public inputs of the VP.

**Comment by XuyangSong** [comment]
> `retrieveVP()` Solvers need information about the VPs in order to be able to satisfy them efficiently. There was some design question around that, was it resolved?

Not yet.
Solvers would get some plaintext of the (intent)notes to generate proofs when receiving the unbalanced partial transaction. We need a design to get the full vk and an understandable description of the (intent)vp from the compressed vk in the note.
